File: server/test/alda/parser/comments_test.clj
(ns alda.parser.comments-test
  (:require [clojure.test :refer :all]
            [alda.parser  :refer (parse-input)]))

(def expected
  '(alda.lisp/score
     (alda.lisp/part {:names ["piano"]}
       (alda.lisp/note (alda.lisp/pitch :c))
       (alda.lisp/note (alda.lisp/pitch :e)))))

(deftest short-comment-tests
  (testing "a short comment"
    (is (= expected (parse-input "piano: c
                                  # d
                                  e")))
    (testing "at the end of a line"
      (is (= expected (parse-input "piano: c # d
                                    e")))
      (testing "without a leading space"
        (is (= expected (parse-input "piano: c#d\ne")))))))

File: client/parser/comments_test.go
package parser

import (
	"testing"

	"alda.io/client/model"
	_ "alda.io/client/testing"
)

func TestComments(t *testing.T) {
	executeParseTestCases(
		t,
		parseTestCase{
			label: "simple comment",
			given: `piano: c
			# d
			e`,
			expect: []model.ScoreUpdate{
				model.PartDeclaration{Names: []string{"piano"}},
				model.Note{Pitch: model.LetterAndAccidentals{NoteLetter: model.C}},
				model.Note{Pitch: model.LetterAndAccidentals{NoteLetter: model.E}},
			},
		},
		parseTestCase{
			label: "comment at the end of a line",
			given: `piano: c # d
			e`,
			expect: []model.ScoreUpdate{
				model.PartDeclaration{Names: []string{"piano"}},
				model.Note{Pitch: model.LetterAndAccidentals{NoteLetter: model.C}},
				model.Note{Pitch: model.LetterAndAccidentals{NoteLetter: model.E}},
			},
		},
		parseTestCase{
			label: "comment without a leading space",
			given: `piano: c #d
			e`,
			expect: []model.ScoreUpdate{
				model.PartDeclaration{Names: []string{"piano"}},
				model.Note{Pitch: model.LetterAndAccidentals{NoteLetter: model.C}},
				model.Note{Pitch: model.LetterAndAccidentals{NoteLetter: model.E}},
			},
		},
	)
}
